src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs

```diff
@@ -160,6 +161,37 @@ private bool GetReadOnly(PropertyInfo property)
             return property.GetSetMethod() == null;
         }
 
+        private bool GetRequired(PropertyInfo property)
+        {
+            var publicCtor = property.DeclaringType?.GetConstructors().Where(c => c.IsPublic).OrderBy(c => c.GetParameters().Count()).FirstOrDefault();
+            if (publicCtor == null)
+            {
+                //ReferenceTypes for inheritance do not have public constructors, and currently there are ResourceData and TrackedResourceData.
+                //We treat Id, Name, ResourceType, Location in them as required.
+                if (property.DeclaringType!.Name.Equals("ResourceData", StringComparison.Ordinal) || property.DeclaringType!.Name.Equals("TrackedResourceData", StringComparison.Ordinal))
+                {
+                    if (property.Name == "Id" ||
+                        property.Name == "Name" ||
+                        property.Name == "ResourceType" ||
+                        property.Name == "Location")
+                        return true;
+                }
+                return false;
+            }
+            return publicCtor.GetParameters().Any(param => param.Name?.Equals(property.Name, StringComparison.OrdinalIgnoreCase) == true && param.GetType() == property.GetType());
+        }
```

Comment:
I still do not think this makes sense - taking ResourceData and TrackedResourceData as an example: since they could match with an optional of SystemData, should this mean everything except systemData is required?
This could work for all its properties - except for `Tags`, but it is a collection, what would it change if it is required? I believe nothing, since it has an initializer and always has a value when serialized

Reply:
The serialization code for the derived classes of ResourceData and TrackedResourceData will change based on whether the property is required. For instance, if Tags is treated as required, the serializer will simply write the tag property and values without checking whether it exists in the requests. Services handle null values differently, so it might cause issue with the previous assumption that everything is required.